src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -5869,6 +5869,22 @@ static QVariant fcnColorRgb( const QVariantList &values, const QgsExpressionCont
   return QStringLiteral( "%1,%2,%3" ).arg( color.red() ).arg( color.green() ).arg( color.blue() );
 }
 
+static QVariant fcnColorRgbF( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  const float red = static_cast<float>( QgsExpressionUtils::getDoubleValue( values.at( 0 ), parent ) );
+  const float green = static_cast<float>( QgsExpressionUtils::getDoubleValue( values.at( 1 ), parent ) );
+  const float blue = static_cast<float>( QgsExpressionUtils::getDoubleValue( values.at( 2 ), parent ) );
+  const float alpha = static_cast<float>( QgsExpressionUtils::getDoubleValue( values.at( 3 ), parent ) );
+  QColor color = QColor::fromRgbF( red, green, blue, alpha );
```

Comment:
I'd suggest clamping the float values to 0-1, otherwise qt will spam a billion messages to the console and effectively freeze qgis if an out-of-range value is passed to the function

Reply:
Shall we not detect it and report as an evaluation error, so user can fix its mistake instead of silently creating clamped color?

Reply:
@troopa81 , I'd be -1 to making an out of range value an evaluation error. It can sometimes happen with vector tile layers having complex expressions converted from mapboxgl. I think we're much better being tolerant here, and clamping the value to what's accepted. It'd also just break away from a well established behavior of color function tolerance on that front.

For people who are developing expressions, we can assume they'll be looking for expected color outcomes and so the visual feedback will be the "reporting" here :)

Reply:
@troopa81 , thanks.